Spigot-Server-Patches/0173-Add-hit-entity-to-ProjectileHitEvent.patch

```diff
@@ -0,0 +1,108 @@
+@@ -459,7 +459,7 @@ public class EntityFishingHook extends Entity {
+                     if (playerFishEvent.getExpToDrop() > 0) {
+                         this.owner.world.addEntity(new EntityExperienceOrb(this.owner.world, this.owner.locX, this.owner.locY + 0.5D, this.owner.locZ + 0.5D, playerFishEvent.getExpToDrop()));
+                     }
+-                    // CraftBukkit end                
++                    // CraftBukkit end
+                 }
+ 
+                 i = 1;
+diff --git a/src/main/java/net/minecraft/server/EntityProjectile.java b/src/main/java/net/minecraft/server/EntityProjectile.java
+index 770f130..e95cc24 100644
+--- a/src/main/java/net/minecraft/server/EntityProjectile.java
++++ b/src/main/java/net/minecraft/server/EntityProjectile.java
+@@ -175,7 +175,7 @@ public abstract class EntityProjectile extends Entity implements IProjectile {
+                 this.a(movingobjectposition);
+                 // CraftBukkit start
+                 if (this.dead) {
+-                    org.bukkit.craftbukkit.event.CraftEventFactory.callProjectileHitEvent(this);
++                    org.bukkit.craftbukkit.event.CraftEventFactory.callProjectileHitEvent(this, movingobjectposition.entity); // Paper - include Entity in movingobjectposition
+                 }
+                 // CraftBukkit end
+             }
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index e39de2b..c92a80b 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -197,7 +197,7 @@ public class CraftEventFactory {
+     public static PlayerInteractEvent callPlayerInteractEvent(EntityHuman who, Action action, BlockPosition position, EnumDirection direction, ItemStack itemstack, EnumHand hand) {
+         return callPlayerInteractEvent(who, action, position, direction, itemstack, false, hand);
+     }
+-    
++
+     public static PlayerInteractEvent callPlayerInteractEvent(EntityHuman who, Action action, BlockPosition position, EnumDirection direction, ItemStack itemstack, boolean cancelledBlock, EnumHand hand) {
+         Player player = (who == null) ? null : (Player) who.getBukkitEntity();
+         CraftItemStack itemInHand = CraftItemStack.asCraftMirror(itemstack);
+@@ -652,7 +652,7 @@ public class CraftEventFactory {
+         if (!event.isCancelled()) {
+             state.update(true);
+         }
+-        
++
+         return !event.isCancelled();
+     }
+ 
+@@ -782,6 +782,15 @@ public class CraftEventFactory {
+         return event;
+     }
+ 
++    // Paper start
++    public static ProjectileHitEvent callProjectileHitEvent(Entity entity, Entity hitEntity) {
++        ProjectileHitEvent event = new ProjectileHitEvent((Projectile) entity.getBukkitEntity(), (hitEntity == null ? null : (org.bukkit.entity.Entity) hitEntity.getBukkitEntity()));
```

Comment:
If you have a look at CraftEventFactory, there are several org.bukkit.entity.Entity references as the current scope is at the NMS entity, not the bukkit one.